File: guides/micronaut-security-session-database-authentication/java/src/main/java/example/micronaut/constraints/PasswordMatchMessages.java
/*
 * Copyright 2017-2023 original authors
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package example.micronaut.constraints;

import io.micronaut.context.StaticMessageSource;
import jakarta.inject.Singleton;

@Singleton // <1>
public class PasswordMatchMessages extends StaticMessageSource {

    public static final String PASSWORD_MATCH_MESSAGE = "Passwords do not match";

    private static final String MESSAGE_SUFFIX = ".message";

    public PasswordMatchMessages() {
        addMessage(PasswordMatch.class.getName() + MESSAGE_SUFFIX, PASSWORD_MATCH_MESSAGE);
    }
}

File: guides/micronaut-security-session-database-authentication/python/src/example/micronaut/constraints/password_match_messages.py
from jakarta.inject import Singleton
from micronaut.context import MessageSource, StaticMessageSource
from micronaut.context.annotation import Factory

from .password_match import MESSAGE


@Factory
class PasswordMatchMessages:
    PASSWORD_MATCH_MESSAGE = "Passwords do not match"

    @Singleton  # <1>
    def message_source(self) -> MessageSource:
        source = StaticMessageSource()
        source.addMessage(MESSAGE, self.PASSWORD_MATCH_MESSAGE)
        return source
